lib/vmpooler/api/v1.rb

```diff
@@ -542,8 +542,13 @@ def need_token!
                 backend.hset('vmpooler__vm__' + params[:hostname], param, arg)
               when 'tags'
                 arg.keys.each do |tag|
-                    backend.hset('vmpooler__vm__' + params[:hostname], 'tag:' + tag, arg[tag])
-                    backend.hset('vmpooler__tag__' + Date.today.to_s, params[:hostname] + ':' + tag, arg[tag])
+                  if Vmpooler::API.settings.config[:tagfilter] and Vmpooler::API.settings.config[:tagfilter][tag]
+                    filter = Regexp.new(Vmpooler::API.settings.config[:tagfilter][tag])
```

Comment:
I am not sure the expense of creating new Regexp objects. This should be moved to a hash that is either updated when new filters are found or at app initialization.

```
regexes[tag] ||= Regexp.new(...)
arg[tag] = arg[tag].gsub(regexes[tag], '')
```